src/Validation.Common.Job/Storage/ValidatorStateService.cs

```diff
@@ -59,6 +59,30 @@ public async Task<ValidatorStatus> GetStatusAsync(INuGetValidationRequest reques
             return status;
         }
 
+        public async Task<ValidatorStatus> GetStatusAsync(IValidationRequest request)
+        {
+            var status = await GetStatusAsync(request.ValidationStepId);
+
+            if (status == null)
+            {
+                return new ValidatorStatus
+                {
+                    ValidationId = request.ValidationStepId,
+                    ValidatorName = _validatorName,
+                    State = ValidationStatus.NotStarted,
+                    ValidatorIssues = new List<ValidatorIssue>(),
+                };
+            }
+            else if (status.ValidatorName != _validatorName)
+            {
+                throw new ArgumentException(
+                    $"Validation expected validator {status.ValidatorName}, actual {_validatorName}",
```

Comment:
I think this is flipped. Should this be swapped?

```suggestion
                    $"Validation expected validator {_validatorName}, actual {status.ValidatorName}",
```

This `ValidatorStateService` is for `_validatorName` only, whereas `status.ValidatorName` is what the user requested through the `IValidationRequest`.